server/command.go

```diff
@@ -317,6 +327,58 @@ func (p *Plugin) executeShowCommand(args *model.CommandArgs) (*model.CommandResp
 	return &model.CommandResponse{}, nil
 }
 
+func (p *Plugin) executeSyncCommand(args *model.CommandArgs, parameters []string) (*model.CommandResponse, *model.AppError) {
+	if !p.API.HasPermissionTo(args.UserId, model.PermissionManageSharedChannels) {
+		return p.cmdError(args.UserId, args.ChannelId, "Unable to execute the command, you require `manage_shared_channels` permission to sync the channel.")
+	}
+
+	channel, appErr := p.API.GetChannel(args.ChannelId)
+	if appErr != nil {
+		return p.cmdError(args.UserId, args.ChannelId, "Unable to get the current channel information.")
+	}
+
+	if channel.Type == model.ChannelTypeDirect || channel.Type == model.ChannelTypeGroup {
+		return p.cmdError(args.UserId, args.ChannelId, "Syncrhonizing DMs or GMs is not supported yet")
+	}
+
+	hours := 24
+	if len(parameters) == 1 {
+		var err error
+		hours, err = strconv.Atoi(parameters[0])
+		if err != nil {
+			return p.cmdError(args.UserId, args.ChannelId, "Invalid command, please pass a valid number of hours as a parameter.")
+		}
+	} else if len(parameters) > 1 {
+		return p.cmdError(args.UserId, args.ChannelId, "Invalid command, please pass the number of hours as a parameter.")
+	}
+
+	link, err := p.store.GetLinkByChannelID(args.ChannelId)
+	if err != nil || link == nil {
+		return p.cmdError(args.UserId, args.ChannelId, "Link doesn't exist.")
```

Comment:
This error isn't super clear: should be say something like, "This is not a shared channel?"

Reply:
The error is consistent with the rest of the code, and functionality, there is a "link" command to link a channel, so "shared channels" can be more confusing here than clear.

I would suggest changing it to something like "This channel is not linked to a MSTeams channel.", to something like that.